lucene/core/src/java/org/apache/lucene/document/VectorField.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.lucene.document;
+
+
+import org.apache.lucene.codecs.VectorValues;
+
+/**
+ * Per-document vector value; {@code float} array with indexed knn graph for fast approximate nearest neighbor search.
+ */
```

Comment:
knn graph ?

Reply:
I will clean up these comments. It's clear that I copied some of Tomoko and Mike's great work :)